sktime/forecasting/compose/_column_ensemble.py

```diff
@@ -0,0 +1,169 @@
+
+from sklearn.base import clone
+
+from sktime.forecasting.base._base import DEFAULT_ALPHA
+from sktime.forecasting.base._base import BaseForecaster
+from sktime.forecasting.base._meta import _HeterogenousEnsembleForecaster
+
+
+class ColumnEnsembleForecaster(_HeterogenousEnsembleForecaster):
+    """Forecast each series with separate forecaster.
+
+    Parameters
+    ----------
+    forecasters : list of tuples (str, estimator, int or str)
+        With name as str, estimator as sktime-like estimator,
+        index as str or int
+
+    Examples
+    --------
+    >>> from sktime.forecasting.compose import ColumnEnsembleForecaster
+    >>> from sktime.forecasting.exp_smoothing import ExponentialSmoothing
+    >>> from sktime.forecasting.trend import PolynomialTrendForecaster
+    >>> from sktime.datasets import load_longley
+    >>> _, y = load_longley()
+    >>> y = y.drop(columns=["UNEMP", "ARMED", "POP"])
+    >>> forecasters = [("trend", PolynomialTrendForecaster(), 0),\
+                        ("ses", ExponentialSmoothing(trend='add'), 1)]
+    >>> forecaster = ColumnEnsembleForecaster(forecasters=forecasters)
+    >>> forecaster.fit(y, fh=[1, 2, 3])
+    ColumnEnsembleForecaster(...)
+    >>> y_pred = forecaster.predict()
+    """
+
+    _required_parameters = ["forecasters"]
+    _tags = {
+        "scitype:y": "both",
+        "univariate-only": False,
+        "y_inner_mtype": "pd.DataFrame",
+        "requires-fh-in-fit": False,
+        "handles-missing-data": False,
+    }
+
+    def __init__(self, forecasters):
+        self.forecasters = forecasters
+        super(ColumnEnsembleForecaster, self).__init__(forecasters=forecasters)
+
+    @property
+    def _forecasters(self):
+        """Make internal list of forecasters.
+
+        The list only contains the name and forecasters, dropping
+        the columns. This is for the implementation of get_params
+        via _HeterogenousMetaEstimator._get_params which expects
+        lists of tuples of len 2.
+        """
+        return [(name, forecasters) for name, forecasters, _ in self.forecasters]
+
+    @_forecasters.setter
+    def _forecasters(self, value):
+        self.forecasters = [
+            (name, forecasters, columns)
+            for ((name, forecasters), (_, _, columns)) in zip(value, self.forecasters)
+        ]
+
+    def _fit(self, y, X=None, fh=None):
+        """Fit to training data.
+
+        Parameters
+        ----------
+        y : pd.DataFrame
+            Target time series to which to fit the forecaster.
+        fh : int, list or np.array, optional (default=None)
+            The forecasters horizon with the steps ahead to to predict.
+        X : pd.DataFrame, optional (default=None)
+            Exogenous variables are ignored.
+
+        Returns
+        -------
+        self : returns an instance of self.
+        """
+        names, forecasters, indices = self._check_forecasters()
+        if len(set(indices)) != len(indices):
```

Comment:
One last comment, perhaps the index checks can go into the `_check_forecasters` method?